.github/workflows/docs-preview.yml

```diff
@@ -0,0 +1,89 @@
+name: 'Documentation (Preview)'
+
+# INFO:
+# This workflow automates generating previews of our docs for PRs.
+# For security reasons, it is necessary to split the workflow into two separate jobs to manage trust safely.
+
+# MAINTAINERS:
+# - This set of `pull_request_target` + `workflow_call` workflows replaces the prior `pull_request` (untrusted) + `workflow_run` (trusted) workflows approach
+#   due to the need for acquiring PR metadata (Head SHA + PR number), which the `workflow_run` event context cannot provide when PRs are from forks.
+# - Please be mindful of the risks when maintaining this workflow:
+#   https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+
+on:
+  pull_request_target:
+    paths:
+      - 'docs/**'
+      - '.github/workflows/scripts/docs/build-docs.sh'
+
+# If this workflow is triggered while an earlier instance is already running for the PR,
+# cancel that instance in favor of this newly triggered run:
+concurrency:
+  group: deploypreview-pullrequest-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+# Common inputs shared between the two workflow jobs (`preview` + `deploy`).
+# Composed as JSON to pass as a single input which each called job will map into separate ENV for use.
+env:
+  PREVIEW_CONTEXT: |
+    {
+      "build_dir": "docs/site/",
+      "netlify": {
+        "site_name":     "dms-doc-previews",
+        "deploy_prefix": "pullrequest-${{ github.event.pull_request.number }}"
+      },
+      "pull_request": {
+        "head_repo": "${{ github.event.pull_request.head.repo.full_name }}",
+        "head_sha":  "${{ github.event.pull_request.head.sha }}",
+        "number":    "${{ github.event.pull_request.number }}"
+      }
+    }
+
+# Grant `secrets.GITHUB_TOKEN` only the minimum permissions needed by actions to function:
+# https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token
+# NOTE: See the associated `preview` and `deploy` workflows called for when these permissions are needed.
+permissions:
+  contents: read
+  pull-requests: write
```

Comment:
These are the only "dynamic" config for the workflow. Not sure if the `repo.full_name` context could be used here for manipulation?

`permissions` has been limited accordingly. The `prepare` workflow that later runs the PR branch to build the docs only has `contents: read` available, but AFAIK should not have access to `GITHUB_TOKEN` on the PR author's side of things and they no longer have influence over the workflow itself.

That only leaves `build-docs.sh` as a concern, but with the restricted permissions for `GITHUB_TOKEN` and preventing any secrets from being in scope via `workflow_call` trigger, I think it's very close to the restricted environment of `pull_request` event triggered workflows? (_provided any future additions don't introduce risks in `docs-preview-prepare.yml`_)